src/main/java/org/usfirst/frc4904/robot/RobotMap.java

```diff
@@ -31,50 +67,101 @@
+            public static final double D_TOLERANCE = -1;
         }
 
     }
 
+    public static class DriveConstants {// TODO: Change Constants
+        // Field Carpet
+        // public static final double ksVolts = 0.0018;
+        // public static final double kvVoltSecondsPerMeter = 4.9;
+        // public static final double kaVoltSecondsSquaredPerMeter = 0.184;
+        // public static final double kTrackwidthMeters = .61; //0.5842
+        // public static final double kPDriveVel = 6.27;
+        // School Carpet
+        public static final double VOLTS = -1;
+        public static final double VOLT_SECONDS_PER_METER = -1;
+        public static final double VOLT_SECONDS_SQUARED_PER_METER = -1;
+        public static final double TRACK_WIDTH_METERS = -1; // 0.5842 // 0.6063751884752512
+        public static final double DRIVE_VEL = -1;
+        public static final DifferentialDriveKinematics DRIVE_KINEMATICS = new DifferentialDriveKinematics(
+                TRACK_WIDTH_METERS);
+        public static final SplineDriveConstants DRIVE_CONSTANTS = new SplineDriveConstants(VOLTS,
+                VOLT_SECONDS_PER_METER, VOLT_SECONDS_SQUARED_PER_METER, TRACK_WIDTH_METERS, DRIVE_VEL);
+    }
+
+    public static class AutoConstants {// TODO: Change Constants
+        public static final double MAX_SPEED_METERS_PER_SECOND = -1;
+        public static final double MAX_ACCELERATION_METERS_PER_SECOND_SQUARED = -1;
+        public static final double RAMSETE_B = -1;
+        public static final double RAMSETE_ZETA = -1;
+        public static final SplineAutoConstants AUTO_CONSTANTS = new SplineAutoConstants(MAX_SPEED_METERS_PER_SECOND,
+                MAX_ACCELERATION_METERS_PER_SECOND_SQUARED, RAMSETE_B, RAMSETE_ZETA);
+    }
+
     public static class Component {
+        public static PDP pdp;
+        public static Motor leftDriveA;
+        public static Motor leftDriveB;
+        public static Motor rightDriveA;
+        public static Motor rightDriveB;
+        public static TankDriveShifting chassis;
+        public static SolenoidShifters shifter;
+        public static EnableableModifier leftWheelAccelerationCap;
+        public static EnableableModifier rightWheelAccelerationCap;
+        public static CANCoder leftWheelEncoder;
+        public static CANCoder rightWheelEncoder;
+        public static EncoderPair chassisEncoders;
+        public static CANCoderConfiguration canCoderConfiguration;
+        public static NavX navx;
+        public static CustomPIDController turnPID;
+        public static SensorDrive splinesChassis;
     }
 
     public static class Input {
     }
 
+    public static class NetworkTables {
+        public static NetworkTableInstance inst;
+        public static NetworkTable table;
+
+        public static class Vision {
+            public static NetworkTable table;
+            public static NetworkTableEntry xDistanceToTarget;
+            public static NetworkTableEntry yDistanceToTarget;
```

Comment:
To try to avoid confusion we've taken to calling these `distance` (planar distance from target) and `height` (vertical distance from target).